### PR TITLE
Feature/autoencode

### PR DESCRIPTION
**Pull request by eschnou**
Autoencode is useful to debug things and validate that the pipeline is properly working, it is also quite fun to play with. Using this added option, the network is just feeding the same value as input and output. Results on the validation set are surprisingly good, even with small hidden layer size and embedding size.

Also bundled in this pull request is the addition of logging during training of the evolution of the loss function and perplexity.

As always, feel free to reject, edit, or merge if you think it can be useful to others. Let me know if you have comments and want me to change anything.

### CODE REVIEW

**Comment by Conchylicultor** [comment]
I just tried and it works great. Thanks again for your contribution.

**Comment by julien-c** [comment]
Sounds great! Care to share a trained model and/or sample results?
